### DISCUSSION TITLE
Resetting Totals

### DISCUSSION THREAD

**Discussion by matthew270492** [nathanmarlor/foxess_modbus#682] [Q&A] [answered]
I have changed my batteries from HV2600 to EP11 running through the same inverter. On the Fox App the total charge & Total discharge have reset with the new battery however HA hasn't and is continuing to increase. 

H1- 3.7 using RS485

**Comment by FozzieUK** [comment]
Which inverter is it, and how are you connected to the inverter RS485 or by LAN ?

**Reply by matthew270492** [reply]
H1 3.7 and via RS485

**Reply by FozzieUK** [reply]
Can you do a register read to see what the inverter is reporting.

Go to Developer Tools and Actions (or Services as it used to be called) and select Foxess - Modbus Read registers

If you could do a read on register 11075 with a count of 2 (similar to this screen)

![IMG_1658](https://github.com/user-attachments/assets/fc190ad2-a98e-4026-b577-6b9419acbd8b)

Click ‘Perform Action’ and post back what the response is please

**Reply by matthew270492** [reply]
![Screenshot_2024-09-29-21-38-50-53_c3a231c25ed346e59462e84656a70e50](https://github.com/user-attachments/assets/8c18edbd-9fcb-47dd-a8bc-1ccfb8c2bcc5)

This is what I got. Thanks for your help.

**Reply by FozzieUK** [reply, marked as answer]
Thanks, 

I had a suspicion this might be the case - the modbus integration is reading the total discharge and charge registers from the inverter, but these are specific to the inverter - in your example above the response is a reading of 1268.5kWh for the total battery discharge recorded by your inverter.

The individual battery discharge and charge totals registers are not exposed over modbus, i've just checked the latest modbus spec sent by Fox this weekend and that doesn't show battery specific registers either and so this isn't something we have access to, it must be something managed in the Fox cloud.

I've noted that original V1 batteries do not provide this information, so this appears to be a newer feature of V2 (or even just EP) firmware versions (perhaps some other V2 battery owners HV/Mira/ECS can check this in their app)

This is what the app says for my V1 HV2600's
![b1](https://github.com/user-attachments/assets/b27b4f28-f44e-41a5-9249-addaa53eb14a)

But it would be easy enough to replicate the function by creating 2 helpers in HA to hold the previous battery totals, and then use those as a base to create 2 sensors to hold your new battery totals going forwards.

**Reply by matthew270492** [reply]
Thanks for looking into it. My App has the same info as yours so if needed I can get it from there. I sometimes look at the history which the Fox app cannot do.
